lib/validator.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+// let validator = module.exports = {};
+
+class Validator {
+  constructor(input, rules) {
+    this.input = input;
+    this.rules = rules;
+  }
+  isValid = () => {
+    return true;
+  };
+  isString = () => {
+    return typeof input === 'string';
+  };
+  isNumber = () => {
+    return typeof input === 'number';
+  };
+
+  isArray = () => {
+    let isInputOfType = array.isArray(input);
+    let isValueOfType = true;
+
+    if (valueType) {
+      for (var i = input.length - 1; i >= 0; i--) {
+        if (typeof input[i] === valueType)
+          continue;
+        else
+          isValueOfType = false;
```

Comment:
Line 20 -29 see if you can make it dryer (if you can not at this point it okay).